common/components/forms/LinkEntryModal.jsx

```diff
@@ -28,21 +35,41 @@ class LinkEntryModal extends React.PureComponent<Props,State> {
 
   constructor(props: Props): void {
     super(props);
+    const formFields: FormFields = {
+      project_link: "",
+    };
+    const validations: $ReadOnlyArray<Validator<FormFields>> = [
+      {
+        checkFunc: (formFields: FormFields) => url.isValidUrl(
+          formFields["project_link"]),
```

Comment:
```suggestion
        checkFunc: (formFields: FormFields) => formFields["project_link"] && url.isValidUrl(
          formFields["project_link"]),
```

We also need to check for blank url's here

Reply:
isURL does not accept blank URL. Let me re-add that test case to the test suite.

Reply:
Actually I'm not sure whether formFields["project_link"] starts out as a blank string or null, but I do know is that in its current state the Link Entry Modal will let me submit a blank url, which is not something we want.  However you want to handle the check is fine; you might leverage _.isEmpty to check for both blank/null in this case.

Reply:
There is a little race-condition here that I identified.
- Event 1: setState re-assigning formFields with `""`
- Event 2: formIsValid gets updated when modal is reset (i.e closing and opening the modal)
Previously, when the form is closed, setState did not get around to empty out formFields before the modal is open again, thus formIsValid still remains true with the previous formField value.
With my latest commit, whenever the modal gets reset, the formIsValid is updated **after** setState has finished re-assigning formFields.
Thanks for catching this :)